### DISCUSSION TITLE
clarification on max_buffer_size vs max_queue_size

### DISCUSSION THREAD

**Discussion by xiankaing** [zendesk/ruby-kafka#873] [Q&A] [answered]
Are these two just the same thing but 
max_queue_size is for async producer
and
max_buffer_size is for normal/sync producer?

**Comment by dasch** [comment, marked as answer]
The async producer in fact _wraps_ a sync producer, so when producing async a message will _first_ be placed in the queue (which is thread safe), then be picked up by a background thread and placed in the sync producer's buffer, and then flushed to Kafka at some point.
